File: BiddingSystem/ClientApp/src/components/Offer.js
import React, { useEffect, useState } from 'react';
import { useNavigate } from "react-router-dom";

const Offer = (props) => {

    const navigate = useNavigate();

    const showOfferDetails = () => {
        navigate("/offer/" + props.offer.id, {id: props.offer.id});
    }

    return (
        <div className="d-flex justify-content-between flex-row p-2 border rounded m-2">
            <div>
                <h3>{props.offer.title}</h3>
                <p>{props.offer.description}</p>
            </div>
            <div>
                <div><b>{props.offer.price} EUR</b></div>
                <button onClick={showOfferDetails} className="btn btn-primary">Details</button>
            </div>
        </div>
    )
}

export default Offer;

File: BiddingSystem/Models/Offer.cs
using System;
namespace BiddingSystem.Models
{
    public class Offer
    {
        public Offer(string title, string description, double price, string password)
        {
            Title = title;
            Description = description;
            Price = price;
            Password = password;
            IsOpen = true;
        }

        public int Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public double Price { get; set; }

        public string? Password { get; set; }

        public bool IsOpen { get; set; }
    }
}

